src/scripts/generateChangelog.ts

```diff
@@ -0,0 +1,125 @@
+import fs from 'fs';
+import {
+  generateMarkDown,
+  getCurrentGitTag,
+  getGitDiff,
+  getLastGitTag,
+  loadChangelogConfig,
+  parseCommits,
+} from 'changelogen';
+import ChangelogenConfig from '../../changelog.config';
+
+/**
+ * Generates a changelog based on git commits and updates the changelog file.
+ *
+ * This function performs the following steps:
+ * 1. Parses command line arguments to get the version.
+ * 2. Retrieves the current and last git tags.
+ * 3. Loads the changelog configuration.
+ * 4. Gets the git diff between the current and last tags.
+ * 5. Parses the commits from the git diff.
+ * 6. Generates the markdown for the changelog.
+ * 7. Extracts the release notes from the changelog.
+ * 8. Updates the changelog file with the new changelog entries.
+ *
+ * Command line arguments:
+ * - `--version` or `-v` followed by the version number to specify the new version.
+ * - `--version=<version>` to specify the new version.
+ * - If no version is specified, the version will be inferred from the arguments.
+ *
+ * @returns A promise that resolves when the changelog has been generated and the file has been updated.
+ * @throws If the output path in the configuration is invalid.
+ */
+async function generateChangelog(): Promise<void> {
+  // Parse command line arguments
+  const args = process.argv.slice(2);
```

Comment:
From my point of view this function is too big. It could be splitted up to be easier to understand. 

The first step would be to move the logic which fetches the version from the CLI into its own method:

```ts
function getVersionFromCLI() {
  const args = process.argv.slice(2);

  // Get the version from the command line arguments
  let version: string | undefined;

  for (let i = 0; i < args.length; i++) {
    const arg = args[i];
    if (arg === '--version' || arg === '-v') {
      version = args[i + 1];
    } else if (arg.startsWith('--version=')) {
      version = arg.split('=')[1];
    } else if (!version && !arg.startsWith('-')) {
      version = arg;
    }
  }
  return version;
}
```

Calling the function only

```ts
const version: string | undefined = getVersionFromCLI();
```